### PR TITLE
Rp2040 Target Support

### PR DESCRIPTION
**Pull request by gundralaa**
Basic unthreaded runtime support for rp2040 target.
Uses the pico-sdk as a dependency and utilizes the high level [time](https://www.raspberrypi.com/documentation/pico-sdk/high_level.html#pico_time) library to provide timer support.
- changed semaphore util to lf_semaphore due to name collisions
- added lf_rp2040_support files
- modified platform cmake

### CODE REVIEW

File: core/platform/lf_rp2040_support.c [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+* physical actions and interupts outside of the runtime.
+*/
+#ifdef LF_UNTHREADED
+
+/**
+ * Enter a critical section where logical time and the event queue are guaranteed
+ * to not change unless they are changed within the critical section.
+ * this can be implemented by disabling interrupts.
+ * Users of this function must ensure that lf_init_critical_sections() is
+ * called first and that lf_critical_section_exit() is called later.
```
Comment (petervdonovan) [inline core/platform/lf_rp2040_support.c:128]:
```suggestion
 * called first and that lf_enable_critical_section_exit() is called later.
```

Is this what you meant (I'm not sure)? It seems like the function that has to be called later should modify `_lf_num_nested_critical_sections`.

File: core/platform/lf_rp2040_support.c [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    if (_lf_num_nested_critical_sections++ == 0) {
+        critical_section_enter_blocking(&_lf_crit_sec);
+    }
+    return 0;
+}
+
+/**
+ * Exit the critical section entered with lf_lock_time().
+ * @return 0 on success, platform-specific error number otherwise.
+ * TODO: needs to be used sparingly, find a better way for event queue
```
Comment (petervdonovan) [inline core/platform/lf_rp2040_support.c:146]: All of these TODOs are basically the same as FIXMEs in that they probably should be moved out of the code and into GitHub issues before this is merged.

**Review by edwardalee — CHANGES_REQUESTED** [review]
This looks pretty good, but the functions defined need comments. E.g., _lf_clock_now should state that it return the time since boot in nanoseconds.  Note that by convention, comments consist of full sentences with proper punctuation and javadoc style annotations. So, for _lf_clock_now, it should be:

```
/**
 * Write the time since boot in nanoseconds into the variable pointed to by the argument and return 0.
 * @param t Pointer to the variable to write to.
 */

Note the use of the imperative case above. I.e., don't say this:

```
 * Writes the time since boot in nanoseconds into the variable pointed to by the argument and return 0.
```

The above is not a sentence in English.

File: core/CMakeLists.txt [hunk 3]
```diff
@@ -76,8 +76,11 @@ endif()
 
 # Link with thread library
 if(DEFINED LF_THREADED OR DEFINED LF_TRACE)
-    find_package(Threads REQUIRED)
-    target_link_libraries(core PUBLIC Threads::Threads)
+    # FIXME make this more general for other platforms
+    if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Rp2040")
+        find_package(Threads REQUIRED)
+        target_link_libraries(core PUBLIC Threads::Threads)
+    endif()
```
[the remark this answers is not in the document]
Comment (gundralaa) [reply core/CMakeLists.txt]: Removed and added validator implementations for RP2040 that disallows threads for now until the threaded runtime is included to lingua-franca pr checklist.

File: core/platform/lf_rp2040_support.c [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***************/
+
+/**
+ * @brief Raspberry Pi Pico support for the C target of Lingua Franca 
+ * Uses the pico-sdk which targets the lower level peripheral layer.
```
[the remark this answers is not in the document]
Comment (gundralaa) [reply core/platform/lf_rp2040_support.c]: changed

**Review by lhstrh — APPROVED** [review]

**Pull request merged**